File: tournaments/minimalNumberOfCoins/minimalNumberOfCoins.js
function minimalNumberOfCoins(coins, price) {
    let result = 0;
    let index = coins.length - 1;

    coins.sort((a, b) => a - b);

    while (0 <= index && 0 < price) {
        if (coins[index] <= price) {
            price -= coins[index];
            ++result;
        } else {
            --index;
        }
    }

    return result;
}

File: tournaments/minimalNumberOfCoins/minimalNumberOfCoins.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub minimalNumberOfCoins {
    my ($coins, $price) = @_;

    my $index = @$coins - 1;
    my $result = 0;
    my @sortedCoins = sort {$a <=> $b} @$coins;

    while (0 <= $index && 0 < $price) {
        if (@sortedCoins[$index] <= $price) {
            $price -= @sortedCoins[$index];
            ++$result;
        } else {
            --$index;
        }
    }

    return $result;
}
